libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/recenter/RecenterUIComponent.kt

```diff
@@ -66,7 +64,7 @@ internal class MapboxRecenterUIComponent(
     }
 
     private fun observeRecenterButtonState() {
-        lifecycleOwner.lifecycleScope.launch {
+        lifecycleOwner.keepExecutingWhenStarted {
```

Comment:
```suggestion
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Reply:
That's not as simple because `repeatOnLifecycle` is a suspend function, see https://github.com/mapbox/mapbox-navigation-android/pull/5332#discussion_r779770125.

Reply:
An extension on lifecycleOwner could be good, otherwise it seems it would need to be

```suggestion
        lifecycleOwner.lifecycleScope.launch {
            lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
```

🤔 . Could be good to make a UiComponent sharable functions